lib/codeowners/cli/wizards/new_owner_wizard.rb

```diff
@@ -25,24 +28,55 @@ def suggest_fixing(line, new_owner)
 
         private
 
-        def prompt(line, new_owner)
-          ask(<<~QUESTION, limited_to: %w[y r i q])
-            Unknown owner: #{new_owner} for pattern: #{line.pattern}. Add owner to the OWNERS file?
-            (y) yes
-            (r) rename owner
-            (i) ignore owner in this session
-            (q) quit and save
+        def suggest_name_from_owners_list(new_owner)
+          require 'fuzzy_match'
+          search = FuzzyMatch.new(@owners_list.owners)
+          (suggested_owner, dice, _lev) = search.find_with_score(new_owner)
+          return suggested_owner if dice && dice > 0.6
+        end
+
+        def prompt(line, new_owner, suggested_owner)
+          prompt_options = build_prompt_options(suggested_owner)
+          accepted_options = filter_options(suggested_owner, prompt_options)
+          ask(<<~QUESTION, limited_to: accepted_options.keys)
+            #{print_prompt(line, new_owner, suggested_owner)}
+            #{print_options(accepted_options)}
           QUESTION
         end
 
+        def print_prompt(line, new_owner, suggested_owner)
+          prompt = "Unknown owner: #{new_owner} for pattern: #{line.pattern}."
+          prompt += " Did you mean #{suggested_owner}?" if suggested_owner
+          prompt + ' Add owner to the OWNERS file?'
+        end
+
+        def print_options(accepted_options)
+          accepted_options.values.join("\n")
+        end
+
+        def filter_options(suggested_owner, prompt_options)
+          return prompt_options if suggested_owner
+
+          prompt_options.reject { |o, _d| o == 'c' }
+        end
+
+        def build_prompt_options(suggested_owner)
+          {
+            'c' => "(c) correct to #{suggested_owner}",
+            'a' => '(a) add a new owner',
+            'r' => '(r) rename owner',
+            'i' => '(i) ignore owner in this session',
+            'q' => '(q) quit and save'
+          }
+        end
+
         def keep_asking_until_valid_owner
           owner = nil
           loop do
             owner = ask('New owner: ')
             owner = '@' + owner unless owner[0] == '@'
-            break if @owners_list.valid_owner?(owner)
+            break owner if @owners_list.valid_owner?(owner)
```

Comment:
Nice 👍 I think we can also immediately `return owner`, it's seems to me more logical here

Reply:
Thank you.  Replaced it with `return owner`